pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -633,10 +633,38 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 		}
 		glog.Infof("Applied KubeletConfig %v on MachineConfigPool %v", key, pool.Name)
 	}
+	if err := ctrl.cleanUpDuplicatedMC(); err != nil {
+		return err
+	}
 
 	return ctrl.syncStatusOnly(cfg, nil)
 }
 
+// cleanUpDuplicatedMC removes the MC of uncorrected version if format of its name contains 'generated-xxx'.
+// BZ 1955517: upgrade when there are more than one configs, these generated MC will be duplicated
+// by upgraded MC with number suffixed name (func getManagedKeyCtrCfg()) and fails the upgrade.
+func (ctrl *Controller) cleanUpDuplicatedMC() error {
+	generatedKubeletCfg := "generated-kubelet"
```

Comment:
does this cleanup a duplicated kubelet config? The function should be called cleanupDuplicatedKC

Reply:
Only the machine config is duplicated, it only cleans up the machine config. The result of `oc get kubeletconfigs.machineconfiguration.openshift.io` doesn't change.

Reply:
You can create a common function for this in the common/helpers.go file as this is cleaning up MCs and is not specific to ctrcfg or kubeletconfig objects.

Reply:
> You can create a common function for this in the common/helpers.go file as this is cleaning up MCs and is not specific to ctrcfg or kubeletconfig objects.

Didn't add it as a common, the cleanup functions are specific to different Controller.